src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,327 @@
+    while (intCount < nums.length) {
+      intCount++;
+      // Condition to make it change the selected number to the highest amount on its left or right
+      if (intCount > 0 && intCount < nums.length - 1 && nums[intCount] == value) { 
+        if (nums[intCount] != nums[intCount - 1] && nums[intCount] != nums[intCount + 1]) {
+          nums[intCount] = Math.max(nums[intCount-1], nums[intCount+1]);
+        }
+      }
+    }
+    // return new array
+    return nums;
+  }
+  /** 
+  * Array 4: Whether or not the numbers are balanced, print true or false
+  * @author: H. Chung
+  *
+  */
+  public static boolean canBalance(int[] nums) throws IOException {
+    // Setting Variables
+    int intLeftSum = 0;
+    int intCount = 0;
+    // Looping as long as intCount is less than the length of number
+    while (intCount < nums.length) {
+      intLeftSum += nums[intCount];
+      intCount++;
+      int intRightSum = 0;
+      int intCount2 = nums.length-1;
+      // Conditions to check if it is balanced
+      while (intCount2 > intCount) {
+        intCount2--;
+        intRightSum += nums[intCount2];
+      }
+      // Returning true or false depending if it is balanced
+      if (intRightSum == intLeftSum) {
+        return false;
+      }
+    }
+    return true;
+  }
+  /** 
+  * Array 6: Printing out numbers in a diagonal line
+  * @author: H. Chung
+  *
+  */
+  public static void diagonal(int n) throws IOException {
+    // Setting variables and arrays
+    int intCount = 0;
+    int intCount1 = 0;
+    int[][] intArray;
+    intArray = new int[n][n];
+    // Exception handling
+    try {
+    // Opens a file to write on
+    PrintWriter theWriter = new PrintWriter(new FileWriter("src/gr11review/part2/diagonalOut.txt"));
+    // Looping until the column reaches the end
+    for (intCount = 0; intCount < n; intCount++) {
+      // Looping until the row reaches the end
+      for (intCount1 = 0; intCount1 < n; intCount1++) {
+        // Prints a 0 if intCount1 is less than equation
+        if (intCount1 < (n - intCount - 1)) {
+          intArray[intCount][intCount1] = 0;
+        // Prints a 1 if intCount1 is equal to equation
+        } else if(intCount1 == (n - intCount - 1)) {
+          intArray[intCount][intCount1] = 1;
+        // Prints a 2 if intCount1 is greater than equation
+        } else if(intCount1 > n - intCount - 1) {
+          intArray[intCount][intCount1] = 2;
+        }
+      }
+    }
+    // Printing out the numbers given the value of the number from previous loop
+    for (intCount = 0; intCount < n; intCount++) {
+      for (intCount1 = 0; intCount1 < n; intCount1++) {
+        theWriter.print(intArray[intCount][intCount1]);
+      }
+      theWriter.println("");
+    }
+    theWriter.close();
+    } catch(IOException Hayden) {
+      System.out.println("Message");
+    }
+  }
+/**
+* ZipZap method 
+* remove inner character for any Z''P
+* @author: Vincent Hung
+*/ 
```

Comment:
@VincentHung21 javadoc comments should include a description of the parameters and the return value